core/src/lib.rs

```diff
@@ -22,23 +28,37 @@ pub trait DIDResolver {
     }
 }
 
-/// Given the credential type and the credential subject information, create a unissued JSON-LD credential.
-/// In order to become a Verifiable Credential, a data integrity proof must be created for the credential and appended to the JSON-LD document.
-pub fn create_credential(
-    _cred_type: &str,
-    _cred_subject: serde_json::Value,
-) -> Result<serde_json::Value, Box<dyn std::error::Error>> {
-    unimplemented!();
-}
+pub trait DocumentBuilder {
+    /// Given the credential type and the credential subject information, create a unissued JSON-LD credential.
+    /// In order to become a Verifiable Credential, a data integrity proof must be created for the credential and appended to the JSON-LD document.
+    /// this is the default implementation of the `create` method. The `create` method can be overridden to create a custom credential.
+    fn create_credential(
+        &self,
+        cred_type: String, 
+        cred_subject: HashMap<String, Value>,
+        property_set: HashMap<String, Value>,
+        id: &str
+    ) -> Result<VerifiableCredential, Box<dyn std::error::Error>> {
+        let vc = VerifiableCredential::new(CONTEXT_CREDENTIALS,
```

Comment:
This function should return a Credential not a VerifiableCredential. In order to create a VC we would need to create a data integrity proof which is the responsibility of the `create_data_integrity_proof`. In a later PR we can maybe investigate how to ease that process.